modules/stochastic_tools/src/samplers/ActiveLearningMonteCarloSampler.C

```diff
@@ -0,0 +1,89 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ActiveLearningMonteCarloSampler.h"
+#include "Distribution.h"
+
+registerMooseObject("StochasticToolsApp", ActiveLearningMonteCarloSampler);
+
+InputParameters
+ActiveLearningMonteCarloSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Monte Carlo Sampler for active learning with surrogate model.");
+  params.addRequiredParam<dof_id_type>("num_batch",
```

Comment:
Make a `_num_batch` member variable since you seem to doing `getParam` a lot on it.